.github/workflows/deploy.yml

```diff
@@ -0,0 +1,95 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Deploy
+
+on:
+  push:
+    branches: [ master, develop, cicd, '*staging*' ]
+
+permissions:
+  id-token: write # required to use OIDC authentication
+  contents: read # required to checkout the code from the repo
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout 📥
+      uses: actions/checkout@v2
+
+    - name: Use Node.js 16.14 ⚙️
+      uses: actions/setup-node@v1
+      with:
+        node-version: 16.14
+
+    - name: Setup vars 📋
+      id: vars
+      run: |
+        echo ::set-output name=distdir::$(git rev-parse --short ${{ github.sha }})-16.x-dist
+
+    - name: Install 🎯
+      run: yarn install
+
+    - name: Build 🏃
+      run: node_modules/.bin/webpack --config webpack.config.js --env CI --env DIST_BASE_URL='${{ steps.vars.outputs.distdir }}'
+      env:
+        DIST_BASE_URL: ${{ steps.vars.outputs.distdir }}
+    - name: Upload artifacts 📤
+      uses: actions/upload-artifact@v2
+      with:
+        name: build-artifacts
+        path: dist
+        if-no-files-found: error
+
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+    - name: Checkout 📥
+      uses: actions/checkout@v2
+
+    - name: Download artifacts 📥
+      uses: actions/download-artifact@v2
+      with:
+        name: build-artifacts
+        path: dist
+
+    - name: Configure AWS credentials 🔐
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        role-to-assume: arn:aws:iam::632571768781:role/tileviewer-cicd-role
```

Comment:
This is the exact ARN of the role I set up in IAM to allow read/write access to the tile viewer bucket.